demos/wang-2018/src/convert_to_csv.py

```diff
@@ -0,0 +1,107 @@
+import argparse
+import json
+import pandas as pd
+
+
+def convert_to_csv(args):
+    with open(args.input_cells) as f:
+        cells_json = json.load(f)
+    with open(args.input_molecules) as f:
+        molecules_json = json.load(f)
+    with open(args.input_expression_matrix) as f:
+        expression_matrix = json.load(f)
+
+    molecules_df = pd.DataFrame(index=[], columns=["X", "Y", "Gene"])
+    start_at = 0
+    for molecule_type, molecule_arr in molecules_json.items():
+        molecule_type_ids = [str(i) for i in range(start_at, start_at + len(molecule_arr))]
+        start_at += len(molecule_arr)
+        molecule_type_df = pd.DataFrame(index=molecule_type_ids, columns=molecules_df.columns.values.tolist())
+        molecule_type_df['Gene'] = molecule_type
+        molecule_type_df['X'] = [m[0] for m in molecule_arr]
+        molecule_type_df['Y'] = [m[1] for m in molecule_arr]
+
+        molecules_df = pd.concat([molecules_df, molecule_type_df])
+    molecules_df.index = molecules_df.index.rename('molecule_id')
+
+    cell_ids = cells_json.keys()
+    cells_df = pd.DataFrame(index=cell_ids, columns=["X", "Y"])
+    cells_df['X'] = cells_df.apply(lambda row: cells_json[str(row.name)]['xy'][0], axis='columns')
+    cells_df['Y'] = cells_df.apply(lambda row: cells_json[str(row.name)]['xy'][1], axis='columns')
+    cells_df.index = cells_df.index.rename('cell_id')
+
+    segmentations = dict()
+    for cell_id, cell_obj in cells_json.items():
+        segmentations[cell_id] = cell_obj['poly']
+
+    matrix_df = pd.DataFrame(index=cell_ids, columns=list(expression_matrix.keys()))
+    matrix_df.index = matrix_df.index.astype(str)
+    for gene_id in matrix_df.columns.values.tolist():
+        for cell_id in matrix_df.index.values.tolist():
+            try:
+                matrix_df.at[cell_id, gene_id] = expression_matrix[gene_id]['cells'][cell_id] / expression_matrix[gene_id]['max']
```

Comment:
What is going on here? Rescaling?  Shouldn't the data already be properly scaled?